### PR TITLE
enhancement(aws_s3 source): Adhere to errors spec

### LINKED ISSUE #14195: Verify `aws_s3` source
- [x] Verify internal event processing metrics are tested via `assert_X_compliance` test helpers
- [x] Update `component_discarded_events_total` to match the new [instrumentation spec](https://github.com/vectordotdev/vector/blob/master/docs/specs/instrumentation.md#eventsdropped) by adding `reason` to the emitted logs, and `intentional` to emitted logs and metrics
- [x] Verify that the component correctly emits the correct error events per the [component spec](https://github.com/vectordotdev/vector/blob/master/docs/specs/instrumentation.md#events)

### PR DESCRIPTION
**Pull request by spencergilbert**
Closes #14195

- chore(aws_s3 source): Fix typo static name
- chore(aws_s3 source): Update comments and configurable description
- chore(aws_s3 source): Use SOURCE_TAGS const in tests
- enhancement(aws_s3 source): increment dropped events on StreamClosedError

### CODE REVIEW

File: src/sources/aws_s3/sqs.rs [hunk 1]
```diff
@@ -558,9 +555,10 @@ impl IngestorProcess {
                     BatchStatus::Delivered => Ok(()),
                     BatchStatus::Errored => Err(ProcessingError::ErrorAcknowledgement),
                     BatchStatus::Rejected => {
+                        // We should emit `component_discarded_events_total` here, but is count always 1?
```
Comment (spencergilbert) [inline src/sources/aws_s3/sqs.rs:558]: Big question here 💭

Comment (neuronull) [reply src/sources/aws_s3/sqs.rs:558]: I believe you can get the count of items (events) from the stream like this:

```
let (count, _) = stream.size_hint(); 
```

Should that also be used above in `StreamClosedError` ?

Comment (spencergilbert) [reply src/sources/aws_s3/sqs.rs:558]: The `StreamClosedError` _seems_ like it's dealing with a single event at a time, unless the naming is bonkers.

Comment (bruceg) [reply src/sources/aws_s3/sqs.rs:558]: How does event array counts play into this?

Comment (spencergilbert) [reply src/sources/aws_s3/sqs.rs:558]: Yeah, that's a big part of my uncertainty 😅

Comment (spencergilbert) [reply src/sources/aws_s3/sqs.rs]: Based on discussion with @jszwedko and @neuronull, removed this in https://github.com/vectordotdev/vector/pull/14344/commits/19c050683ec00f139b794ace2041800521a49496

File: src/sources/aws_s3/sqs.rs [hunk 2]
```diff
@@ -555,11 +556,6 @@ impl IngestorProcess {
                     BatchStatus::Delivered => Ok(()),
                     BatchStatus::Errored => Err(ProcessingError::ErrorAcknowledgement),
                     BatchStatus::Rejected => {
-                        // We should emit `component_discarded_events_total` here, but is count always 1?
-                        error!(
-                            message = "Sink reported events were rejected.",
-                            internal_log_rate_secs = 10,
-                        );
                         // Failed events cannot be retried, so continue to delete the SQS source message.
```
Comment (neuronull) [inline src/sources/aws_s3/sqs.rs:559]: Should we expand on this comment to mention that we are intentionally not emitting error / events dropped here as it should be handled in the sink ?

Comment (spencergilbert) [reply src/sources/aws_s3/sqs.rs:559]: Probably, it's sort of similar to the note we add around the call to `encoding` that handles the error. I'm not sure if that's better suited for copied comments in each component or something more centralized.

Comment (neuronull) [reply src/sources/aws_s3/sqs.rs:559]: Although it isn't pretty, I am probably of the vote for copied comments in each component. Even a one liner that is enough of a hint to get the curious person to look for the more centralized and detailed comment.
But I defer to your judgement

**Review by neuronull — APPROVED** [review]

**Pull request merged**